clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2401,69 +2410,99 @@ void handleResponse(AbstractResponse abstractResponse) {
-
-                                    for (DescribeGroupsResponse.GroupMember groupMember : members) {
-                                        Set<TopicPartition> partitions = Collections.emptySet();
-                                        if (groupMember.memberAssignment().remaining() > 0) {
-                                            final PartitionAssignor.Assignment assignment = ConsumerProtocol.
-                                                deserializeAssignment(groupMember.memberAssignment().duplicate());
-                                            partitions = new HashSet<>(assignment.partitions());
-                                        }
-                                        final MemberDescription memberDescription =
-                                            new MemberDescription(groupMember.memberId(),
-                                                groupMember.clientId(),
-                                                groupMember.clientHost(),
-                                                new MemberAssignment(partitions));
-                                        memberDescriptions.add(memberDescription);
-                                    }
-                                    final ConsumerGroupDescription consumerGroupDescription =
-                                            new ConsumerGroupDescription(groupId,
-                                                protocolType.isEmpty(),
-                                                memberDescriptions,
-                                                groupMetadata.protocol(),
-                                                ConsumerGroupState.parse(groupMetadata.state()),
-                                                fcResponse.node());
-                                    future.complete(consumerGroupDescription);
-                                }
-                            }
-                        }
+                    returnedGroupId.add(groupId);
 
-                        @Override
-                        void handleFailure(Throwable throwable) {
-                            KafkaFutureImpl<ConsumerGroupDescription> future = futures.get(groupId);
-                            future.completeExceptionally(throwable);
-                        }
-                    }, nowDescribeConsumerGroups);
+                    if (returnedGroupId.size() == groupIds.size()) {
+                        sendGroupRequest();
+                    }
                 }
 
                 @Override
                 void handleFailure(Throwable throwable) {
                     KafkaFutureImpl<ConsumerGroupDescription> future = futures.get(groupId);
                     future.completeExceptionally(throwable);
+
+                    returnedGroupId.add(groupId);
+
+                    if (returnedGroupId.size() == groupIds.size()) {
+                        sendGroupRequest();
+                    }
+                }
+
+                void sendGroupRequest() {
```

Comment:
This method sends multiple requests now. It would be useful to rename it to `sendGroupRequests()`

Reply:
Fixed.